main/src/main/scala/sbt/internal/KeyIndex.scala

```diff
@@ -87,13 +109,34 @@ private[sbt] final class AKeyIndex(val data: Relation[Option[AttributeKey[_]], S
   def tasks: Set[AttributeKey[_]] = data._1s.flatten.toSet
   def tasks(key: String): Set[AttributeKey[_]] = data.reverse(key).flatten.toSet
 }
-private[sbt] final class ConfigIndex(val data: Map[Option[String], AKeyIndex]) {
+
+/*
+ * data contains the mapping between a configuration and keys.
+ * identData contains the mapping between a configuration and its identifier.
```

Comment:
why is this necessary?

Reply:
We can't change the type of `Def.ScopeKey[_]`, which binds to `sbt.Scope`, which includes `sbt.ConfigKey`, wrapper around a `String` that represents the `name` of configuration. The mapping information between the Scala identifier and `String` thus needs to be available during conversion back and forth. For example, if the user types in `IntegrationTest/test`, the new parser needs to parse it as `ConfigKey("it")` and `AttributeKey[Unit]("test")`.